src/map/battle.c

```diff
@@ -3878,6 +3879,8 @@ static struct Damage battle_calc_magic_attack(struct block_list *src, struct blo
 			))
 				flag.imdef = 1;
 		}
+		if (tsd && (i = pc->sub_skillatk_bonus(tsd, skill_id)))
+			ad.damage -= ad.damage * i / 100;
```

Comment:
if here want to calc percent, better use ``get_percentage`` or ``get_percentage64``